code/game/objects/structures/mortar.dm

```diff
@@ -228,8 +244,13 @@
 	"<span class='notice'>You undeploy [src].")
 	playsound(loc, 'sound/items/deconstruct.ogg', 25, 1)
 	new /obj/item/mortar_kit(loc)
+	UnregisterSignal(src, COMSIG_MORTAR_TARGET)
 	qdel(src)
 
+/obj/structure/mortar/obj_destruction(damage_amount, damage_type, damage_flag)
+	UnregisterSignal(src, COMSIG_MORTAR_TARGET)
```

Comment:
Signals are automaticly removed on object destruction
Don't use obj_destruction proc
Ping me on how to do this properly